### PR TITLE
Minor change: Added tif and tiff to white_list_formats

### PR DESCRIPTION
**Pull request by MiliasV**
Hello there,
It just took me some time to understand why "flow_from_directory" didn't work for me, only to realize that the "tif" format was not in the white_list_formats. So, I added it :) .

### CODE REVIEW

**Comment by Dref360** [comment]
This could be an issue for a lot of people. PIL doesn't work if the tiff has more than 3 channels.
https://github.com/python-pillow/Pillow/issues/1888

**Comment by fchollet** [comment]
@Dref360 your call.

**Comment by MiliasV** [comment]
I don't have a counter argument. If you think that it is going to cause more problems than it will solve I understand. Let me know 😄.

**Comment by Dref360** [comment]
@charlesauthier

**Comment by Dref360** [comment]
The only workaround I see is to use Pillow and libtiff instead of PIL. This is a major change so we shouldn't go this way.

According to Charles, if you have RGB-IR data (or any multiband data), the output will be deformed by PIL. 

It won't affect most people. So I would like to merge it, but maybe add a warning in `_count_valid_files_in_directory` if there is a `tiff`. Something along the line of : 
```python
UserWarning('Using \'.tiff\' files with multiple bands will cause distortion.',
                      'Please verify your output.')
```

**Comment by MiliasV** [comment]
Something like that :thinking:  ?

File: keras/preprocessing/image.py [hunk 1]
```diff
@@ -963,6 +963,9 @@ def _recursive_list(subpath):
         for fname in files:
             is_valid = False
             for extension in white_list_formats:
+                if fname.lower().endswith('.tiff'):
```
Comment (Dref360) [inline keras/preprocessing/image.py]: This will warns a tremendous amount times, isn't it?
I would just do a check before returning.

Comment (MiliasV) [reply keras/preprocessing/image.py]: Actually, I am pretty sure it won't (I also ran it). 

See the docs at http://docs.python.org/2/library/warnings.html:

"Repetitions of a particular warning for the same source location are typically suppressed."

Comment (Dref360) [reply keras/preprocessing/image.py]: Good then!

**Comment by Dref360** [comment]
LGTM for me. I don't think we need to test that the warning is raised? (It's sometime a pain with pytest)

**Comment by MiliasV** [comment]
If you ask me, I reckon it's ok :)

File: keras/preprocessing/image.py [hunk 2]
```diff
@@ -963,6 +963,9 @@ def _recursive_list(subpath):
...
+                    warnings.warn('Using \'.tiff\' files with multiple bands will cause distortion.'
+                                  'Please verify your output.')
```
Comment (fchollet) [inline keras/preprocessing/image.py]: Do we really need this warning?

Comment (Dref360) [reply keras/preprocessing/image.py]: IMO yes or we shouldn't accept tiff at all. 
We'll get complaints that some .tiff are not handled correctly.

Comment (fchollet) [reply keras/preprocessing/image.py]: Ok then. The message is missing a space for the period, please fix

Comment (MiliasV) [reply keras/preprocessing/image.py]: Sorry about that. Fixed.

**Comment by fanshius** [comment]
you can temporarily solve this problem by adding 'tif', 'tiff' in the format list in image.py source code

**Pull request merged**